### DISCUSSION TITLE
Getting Started with Seeed Studio XIAO nRF52840 (Sense) | Seeed Studio Wiki

### DISCUSSION THREAD

**Discussion by giscus** [Seeed-Studio/wiki-documents#122] [Q&A]
# Getting Started with Seeed Studio XIAO nRF52840 (Sense) | Seeed Studio Wiki

Getting Started with Seeed Studio XIAO nRF52840 (Sense)

https://wiki.seeedstudio.com/XIAO_BLE/

**Comment by yaccard-fr** [comment]
Hello
I am testing my XIAO nrf52840 with MicroPython v1.19.1-995-g0a3600a9a on 2023-03-31
Y use Thonny
The program :
from machine import Pin
Led_bleu_sur_carte = Pin(06, Pin.OUT)
Led_bleu_sur_carte .value(0)
Is OK (my integrated led on the board has turned blue)
But I can't have any other color on my built-in led (RGB) nor any output to 'on' (I'm testing with an external led and a resistor)
The problem may be that the firmware version is still in an unstable phase?
Best Regards    (your products are really excellent)

**Reply by limengdu** [reply]
Hi, the MicroPython version of XIAO nRF52840 is currently in beta and we have not yet officially released it to the public. Thank you for the feedback you provided, we will release the full version as soon as possible.

**Comment by yaccard-fr** [comment]
Yes
Thank you very much for your answer, I will wait for the final
clarification.
Best regards
Yves

**Comment by libertg** [comment]
Hi,
I am trying to get started with XIAO nRF52840 (Sense).
Following the Getting Started instructions the IDE build fails with the following error:

    exec: "adafruit-nrfutil": executable file not found in $PATH
    Compilation error: exec: "adafruit-nrfutil": executable file not found in $PATH

I copied the failing line and executed from a terminal and it executed fine.

Any help is appreciated,
Thanks j

**Comment by Flamy69** [comment]
Hello, I have two kits xiao nRF52840(not Sense version). And I'm trying to create a battery powered application and my board won't boot up after connecting the battery. I have checked the voltage at the input to the board from the battery and I see 3.8V on it, but the board draws 15mA pulse and won't boot. Then when I connect to the board via USB, it looks like the program has disappeared from the board because it won't start even from USB. When I subsequently reload the application via USB, it starts and if I have the battery connected and disconnect the USB, the board continues to run, but if I disconnect the battery and reconnect it, the board does not start and the program disappears from the board.

**Comment by TaiNiePooNie** [comment]
if you're having problems with I2C on the XIAO NRF52840 sense board using CircuitPython,
@Torin_Halsted figured you can make the I2C bus work if you time.sleep(1) after powering the IMU before initializing I2C.

        time.sleep(1)
        i2c = busio.I2C(board.IMU_SCL, board.IMU_SDA)

link:
https://forum.seeedstudio.com/t/imu-i2c-error-with-circuitpython-using-xiao-ble-sense/265020

**Reply by MatthewJeffson** [reply]
Wow! Thank you for sharing here! I really appreciate it!

**Comment by MatthewJeffson** [comment]
Welcome to our product documentation platform! We value user feedback and contributions, so we couldn’t be more grateful that you can help us improving the documentation.

Hence, if you find obvious errors in the product documentation, or you have additional ideas and insights about using products, please [contribute](https://wiki.seeedstudio.com/Contribution-Guide) to us.

If you need technical support, you can email it to [fae@seeed.cc](mailto:fae@seeed.cc).

If you want to discuss the product, I recommend the [Seeed Studio Forum](https://forum.seeedstudio.com/) and [Seeed Studio Discord Channel](https://discord.gg/eWkprNDMU7)

The "Discussion channel" in Github here is a relatively new addition and may not receive much attention.

Nonetheless, we hope you find it helpful with the help of others. We will clean it up on a monthly basis.

**Comment by chandan9031** [comment]
How can I use UART (RX,TX Pin) of Seeed XIAO nRF52840 with Arduino Rx, TX pin.
Software Serial shows error of non compatibility with board.
I want to use RX,TX pin to receive data and show it to either USB com port of Seeed XIAO nRF52840 or send through its Bluetooth.

**Comment by ElVit** [comment]
Hi, I have successfully added a LiPo Battery to my XIAO BLE board and its working.
But I have not found a way to completly turn off the board if the battery is connected.
Do the XIAO BLE board have already an buildin battery off switch?
Or do I have to add an additional switch between XIAO BLE board and the LiPo Battery to be able to turn off the board?
Would it be enough to add a switch on the (+) Pin of the Battery?

**Reply by MatthewJeffson** [reply]
Hi! For what I know, you should add an additional switch.

**Comment by matsujirushi** [comment]
Hi,
In the Specifications comparison, Seeed Studio XIAO nRF52840's battery charge chip is written as BQ25101.
But Seeed Studio XIAO nRF52840 Schematic says BQ25100.
Which is correct?

**Reply by MatthewJeffson** [reply]
Hi! The battery charge chip is BQ25101. There might be some mistakes on the schematic.
I will inform the electrical engineer.

**Comment by liefyuan** [comment]
In file included from C:\Users\zgs\AppData\Local\Temp\arduino\sketches\7766EB584745E0CAB58DC6445B903A87\sketch\Blink.ino.cpp:1:
C:\Users\zgs\AppData\Local\Arduino15\packages\Seeeduino\hardware\nrf52\1.1.3\cores\nRF5/Arduino.h:65:10: fatal error: Adafruit_USBD_CDC.h: No such file or directory
   65 | #include "Adafruit_USBD_CDC.h"
      |          ^~~~~~~~~~~~~~~~~~~~~
compilation terminated.

exit status 1

Compilation error: exit status 1

**Reply by matsujirushi** [reply]
https://forum.seeedstudio.com/t/freezes-when-using-serial-with-1-1-3/270898/2
